internal/provider/resource_autogen_subdomain_delegation.go

```diff
@@ -97,10 +97,11 @@ func (r *resourceAutogenSubdomainDelegation) Read(ctx context.Context, config ma
 			Detail:   err.Error(),
 		})
 	} else if autogenSubdomain == nil {
-		state["id"] = tftypes.NewValue(tftypes.String, "")
-		state["subdomain_id"] = tftypes.NewValue(tftypes.Number, &subdomainId)
-		state["env"] = tftypes.NewValue(tftypes.String, envName)
-		state["nameservers"] = ns.NameserversToProtov5(types.Nameservers{})
+		diags = append(diags, &tfprotov5.Diagnostic{
```

Comment:
This isn't always true, it could fail to find the autogen_subdomain even with subdomain and env being present.